src/python.cc

```diff
@@ -466,88 +416,65 @@ ModelInstanceState::GetInputTensor(
 
   // Load raw data into input_tensor raw data.
   std::string* data_buffer = input_tensor->mutable_raw_data();
-  const void* input_buffer = nullptr;
-  uint64_t buffer_byte_size = 0;
-  TRITONSERVER_MemoryType input_memory_type = TRITONSERVER_MEMORY_CPU;
+  BackendMemory* input_memory;
+
   int64_t input_memory_type_id = 0;
-  for (size_t j = 0; j < input_buffer_count; ++j) {
-    RETURN_IF_ERROR(TRITONBACKEND_InputBuffer(
-        in, j, &input_buffer, &buffer_byte_size, &input_memory_type,
-        &input_memory_type_id));
-
-    if (input_memory_type == TRITONSERVER_MEMORY_GPU) {
-      RETURN_IF_ERROR(TRITONSERVER_ErrorNew(
-          TRITONSERVER_ERROR_UNSUPPORTED,
-          "failed to get input buffer in CPU memory"));
-    }
-    data_buffer->append((const char*)input_buffer, buffer_byte_size);
-  }
+  RETURN_IF_ERROR(BackendMemory::Create(
+      model_state_->TritonMemoryManager(),
+      {BackendMemory::AllocationType::CPU_PINNED_POOL,
+       BackendMemory::AllocationType::CPU},
+      input_memory_type_id, input_byte_size, &input_memory));
+
+  input_tensor_memories_.push_back(input_memory);
```

Comment:
I don't see where you use input_tensor_memories_.
Also, you are allocating a buffer and then copying into it but later I assume you are then copying from this buffer into a protobuf buffer for sending to the python code. You should just copy directly into the protobuf to avoid the allocation and copy. You can size the protobuf field and then pass a pointer to it into the corrector. We have examples of this in grpc_server.cc I believe.

Reply:
I found this. Thanks for the pointer!